lib/dockerregistry/inventory_test.go

```diff
@@ -2042,13 +2039,187 @@ func TestRunChecks(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		got := sc.RunChecks(edges, test.checks)
+		got := sc.RunChecks(test.checks)
 		err := checkEqual(got, test.expected)
 		checkError(t, err,
 			fmt.Sprintf("checkError: test: %v (Error Tracking)\n", test.name))
 	}
 }
 
+func TestImageSizeCheck(t *testing.T) {
+	srcRegName := reg.RegistryName("gcr.io/foo")
+	destRegName := reg.RegistryName("gcr.io/bar")
+	destRC := reg.RegistryContext{
+		Name:           destRegName,
+		ServiceAccount: "robot",
+	}
+	srcRC := reg.RegistryContext{
+		Name:           srcRegName,
+		ServiceAccount: "robot",
+		Src:            true,
+	}
+	registries := []reg.RegistryContext{destRC, srcRC}
+
+	image1 := reg.Image{
+		ImageName: "foo",
+		Dmap: reg.DigestTags{
+			"sha256:000": {"0.9"}}}
+	image2 := reg.Image{
+		ImageName: "bar",
+		Dmap: reg.DigestTags{
+			"sha256:111": {"0.9"}}}
+
+	const mbToBytes = 1024 * 1024
+
+	var tests = []struct {
+		name       string
+		check      reg.ImageSizeCheck
+		manifests  []reg.Manifest
+		imageSizes map[reg.Digest]int
+		expected   error
+	}{
+		{
+			"Image size under the max size",
+			reg.ImageSizeCheck{
+				MaxImageSize:    1,
+				DigestImageSize: make(reg.DigestImageSize),
+			},
+			[]reg.Manifest{
+				{
+					Registries: registries,
+					Images: []reg.Image{
+						image1,
+					},
+					SrcRegistry: &srcRC},
+			},
+			map[reg.Digest]int{
+				"sha256:000": .5 * mbToBytes,
+			},
+			nil,
+		},
+		{
+			"Image size over the max size",
+			reg.ImageSizeCheck{
+				MaxImageSize:    1,
+				DigestImageSize: make(reg.DigestImageSize),
+			},
+			[]reg.Manifest{
+				{
+					Registries: registries,
+					Images: []reg.Image{
+						image1,
+					},
+					SrcRegistry: &srcRC},
+			},
+			map[reg.Digest]int{
+				"sha256:000": 5 * mbToBytes,
+			},
+			fmt.Errorf("The following images were over the max file size" +
+				" of 1MiB: foo"),
+		},
+		{
+			"Multiple images over the max size",
+			reg.ImageSizeCheck{
+				MaxImageSize:    1,
+				DigestImageSize: make(reg.DigestImageSize),
+			},
+			[]reg.Manifest{
+				{
+					Registries: registries,
+					Images: []reg.Image{
+						image1,
+						image2,
+					},
+					SrcRegistry: &srcRC},
+			},
+			map[reg.Digest]int{
+				"sha256:000": 5 * mbToBytes,
+				"sha256:111": 10 * mbToBytes,
+			},
+			fmt.Errorf("The following images were over the max file size" +
+				" of 1MiB: bar, foo"),
+		},
+		{
+			"Max image size is zero",
```

Comment:
I don't think it's reasonable to allow a max image size of 0... This is a case where we have to guard against misbehaving humans who put in nonsense values.